src/withAxiom.ts

```diff
@@ -101,20 +102,47 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
   };
 }
 
+export type AxiomRequest = NextRequest & { log: Logger };
+export type AxiomMiddleware = (
+  request: AxiomRequest,
+  event: NextFetchEvent
+) => NextMiddlewareResult | Promise<NextMiddlewareResult>;
+
 function withAxiomNextEdgeFunction(handler: NextMiddleware): NextMiddleware {
   return async (req, ev) => {
+    const report = {
+      startTime: new Date().getTime(),
+      ip: req.ip,
+      region: req.geo?.region,
+      host: req.nextUrl.host,
+      method: req.method,
+      path: req.nextUrl.pathname,
+      scheme: req.nextUrl.protocol.replace(':', ''),
+      statusCode: 0,
+      userAgent: req.headers.get('user-agent'),
+    };
+    const axiomRequest = req as AxiomRequest;
```

Comment:
This is missing the logger, isn't it?

Reply:
well, this MR only handles printing the edge report. attach request information to log events is in a different [PR](https://github.com/axiomhq/next-axiom/pull/28), I am still working on it though.